src/components/Filters/FilterModal.tsx

```diff
@@ -1,243 +1,296 @@
-									value={
-										schema.properties
-											? {
-													field,
-													title:
-														(schema.properties[field] as JSONSchema).title ||
-														field,
-											  }
-											: { field }
-									}
-									onChange={({ option }) => setEditField(option.field, index)}
-								/>
-							</Box>
-							{operators.length === 1 && (
-								<Txt py={2} px={3} align="center">
-									{operators[0].label}
-								</Txt>
-							)}
-							{operators.length > 1 && (
-								<Box flex={1} mx={1}>
-									<Select<{ slug: string; label: any }>
-										options={operators}
-										valueKey="slug"
-										labelKey="label"
-										// TODO: Remove this logic and pass the primitive value when this is fixed: https://github.com/grommet/grommet/issues/3154
-										value={operators.find((x) => x.slug === operator)}
-										onChange={({ option }) =>
-											setEditOperator(option.slug, index)
-										}
-									/>
-								</Box>
-							)}
-							<Flex flex={1} alignItems="center">
-								<FilterInput
-									operator={operator}
-									value={value}
-									schema={schema.properties![field] as JSONSchema}
-									onUpdate={(v: any) => setEditValue(v, index)}
-								/>
-							</Flex>
-						</Flex>
+			{signatures.map((signature, index) => (
+				<Flex flexDirection="column" key={(signature?.field || '') + index}>
+					{index > 0 && <Txt my={2}>OR</Txt>}
+					<Flex>
+						<FilterInputs
+							properties={properties || []}
+							signature={signature}
+							schema={schema}
+							onUpdate={(signature) => addSignature(signature, index)}
+						/>
 						{index > 0 && (
-							<DeleteButton
+							<Button
+								color="tertiary.main"
+								ml={2}
 								plain
 								fontSize={1}
 								p={1}
-								onClick={() => removeCompound(index)}
+								onClick={() => removeSignature(index)}
 							>
 								<FontAwesomeIcon icon={faTimes} />
-							</DeleteButton>
+							</Button>
 						)}
-					</RelativeBox>
-				);
-			})}
+					</Flex>
+				</Flex>
+			))}
 			<Button
 				mb={2}
 				mt={4}
 				primary
 				underline
-				onClick={() =>
-					setFilters(filters.concat(SchemaSieve.getCleanEditModel(schema)))
-				}
+				onClick={() => addSignature(newSignature, signatures!.length)}
 			>
 				Add alternative
 			</Button>
 		</Modal>
 	);
 };
+
+interface FilterInputs {
+	properties: PropertyOption[];
+	signature?: Partial<FilterSignature>;
+	schema: JSONSchema;
+	onUpdate: (signature: Partial<FilterSignature>) => void;
+}
+
+const FilterInputs = ({
+	properties,
+	signature,
+	schema,
+	onUpdate,
+}: FilterInputs) => {
+	const [searchTerm, setSearchTerm] = React.useState('');
+	const debouncedSetSearchTerm = debounce(setSearchTerm, 300);
+	const operators = React.useMemo(
+		() => (!!signature?.field ? getOperators(schema, signature.field) : []),
+		[signature?.field, signature?.refScheme],
```

Comment:
Does the `signature?.refScheme` need to be here? B/c I don't see it being used in the useMemo body.
If it does, then let's leave a comment so that we don't forget why.
Does this need to be v ?
```suggestion
		[signature?.field, schema],
```